include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -737,6 +737,9 @@ namespace ignition
           const std::unordered_set<ComponentTypeId> &_types = {},
           bool _full = false) const;
 
+      /// \brief Get the rebuildviews mutex for locking
+      public: std::mutex& GetRebuildViewsMutex();
```

Comment:
I believe this addition will break API, which means that this change should probably target `main` instead of `ign-gazebo4`.

cc @chapulina

Reply:
Oops, I misunderstood the conditions for breaking ABI. This is okay :+1:

Reply:
Talked to Ashton offline, this should be ok from an ABI point of view. Just a small nitpick is that our style asks to avoid the `Get` prefix, so it would be good to rename the function:

```suggestion
      /// \brief Get the rebuildviews mutex for locking
      public: std::mutex &RebuildViewsMutex();
```